src/ivc/cyclefold/support_circuit/mod.rs

```diff
@@ -0,0 +1,161 @@
+use std::{marker::PhantomData, num::NonZeroUsize};
+
+use tracing::*;
+
+use crate::{
+    gadgets::ecc,
+    halo2_proofs::{
+        circuit::{Layouter, SimpleFloorPlanner},
+        halo2curves::{
+            ff::{PrimeField, PrimeFieldBits},
+            CurveAffine,
+        },
+        plonk::{Circuit, Column, ConstraintSystem, Error as Halo2PlonkError, Instance},
+    },
+    ivc::cyclefold::support_circuit::ecc::AssignedPoint,
+    main_gate::RegionCtx,
+};
+
+type EccChip<C> = ecc::EccChip<C, tiny_gate::Gate<<C as CurveAffine>::Base>>;
+
+mod tiny_gate;
+use tiny_gate::{Config as GateConfig, Gate};
+
+#[derive(Default)]
+pub struct SupportCircuit<C: CurveAffine> {
+    _p: PhantomData<C>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    gate_config: GateConfig,
+    instance: Column<Instance>,
+}
+
+impl<C: CurveAffine> Circuit<C::Base> for SupportCircuit<C>
+where
+    C::Base: PrimeFieldBits,
+{
+    type Config = Config;
+    type FloorPlanner = SimpleFloorPlanner;
+
+    fn without_witnesses(&self) -> Self {
+        SupportCircuit { _p: PhantomData }
+    }
+
+    fn configure(meta: &mut ConstraintSystem<C::Base>) -> Self::Config {
+        let instance = meta.instance_column();
+        meta.enable_equality(instance);
+
+        Config {
+            gate_config: Gate::configure(meta),
+            instance,
+        }
+    }
+
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<C::Base>,
+    ) -> Result<(), Halo2PlonkError> {
+        let ecc_chip = EccChip::<C>::new(config.gate_config.clone());
+
+        layouter.assign_region(
+            || "",
+            |region| {
+                let mut ctx = RegionCtx::new(region, 0);
+
+                let [x0, y0, l0, x1, y1, l1, _expected_x, _expected_y] = ecc_chip
+                    .gate
+                    .assign_values_from_instance(&mut ctx, config.instance, 0)
+                    .unwrap();
+
+                trace!("instances assigned");
+
+                let [p0, p1] = [
+                    AssignedPoint::<C> { x: x0, y: y0 },
+                    AssignedPoint::<C> { x: x1, y: y1 },
+                ];
+
+                let num_bits =
+                    NonZeroUsize::new(<C::Base as PrimeField>::NUM_BITS as usize).unwrap();
+
+                let l0_bits = ecc_chip
```

Comment:
here we don't have to convert it to bits. Instead, the prover is going to do the conversion "off-circuit" using `fe_to_fe` when he prepares the main circuit inputs.

This is okay since at the end, the decider will verify everything including the field conversion.

Reply:
I didn't want to take bit values at the primary circuit level for comparison. In fact, we still have to calculate that it is L0 & L1 from gamma, so we have to do these conversions there

I.e. this is probably more optimal, but I want to start with a simple comparison at the value level